app/user.go

```diff
@@ -77,6 +89,10 @@ func (a *App) CreateUserWithHash(user *model.User, hash string, data string) (*m
 
 	a.AddDirectChannels(team.Id, ruser)
 
+	if err := a.DeleteToken(token); err != nil {
```

Comment:
Again problems above will cause the token not to be deleted.

Reply:
If there is any problem above, you don't want to get deleted the token, you only want to delete (invalidate) the token, after the proper use. Any fail try not invalidate the token.